Assets/HoloToolkit/Input/Scripts/BasicCursor.cs

```diff
@@ -48,7 +44,25 @@ protected virtual void Awake()
             meshRenderer.enabled = false;
 
             // Cache the cursor default rotation so the cursor can be rotated with respect to the original orientation.
-            cursorDefaultRotation = this.gameObject.transform.rotation;
+            cursorDefaultRotation = gameObject.transform.rotation;
+        }
+
+        private bool GetGazeManagerReference()
+        {
+            gazeManagerReference = GazeManager.Instance;
+
+            if (gazeManagerReference == null)
+            {
+                return false;
```

Comment:
I'd prefer changing Awake to Start. If you're worried about inherited cases, we could also move only the `if ((GazeManager.Instance.RaycastLayerMask...` etc part into Start and leave the rest of Awake alone.

If we choose to go the route where this code is moved into LateUpdate, I don't think we have to wait any frames before throwing the error. Have you run into a case where the reference is null for a few frames? As soon as GazeManager's GameObject is instantiated (which would happen before this script's Start and way before LateUpdate), FindObjectOfType will return non-null.